### PR TITLE
added email verification and notifications on loginpage

### PR DESCRIPTION
**Pull request by gnorbsl**
Added email verification, also remove the ".." from all file paths in the views to avoid issues with nested routes.

../img would only work for routes like donate-gifts.com/user but not for donate-gifts.com/user/verify

switched email sending from callback based to async based.

added notifications for frontend if something goes wrong during login. Frontend devs should prettify this.

### CODE REVIEW

File: routes/users.js [hunk 1]
```diff
@@ -211,20 +216,27 @@ router.post('/signup', redirectProfile, async (req, res) => {
...
-		var userId = mongoose.Types.ObjectId(newUser._id);
-		req.session.userId = userId;
+		req.session.userId = mongoose.Types.ObjectId(newUser._id);
 		try {
 			await newUser.save();
 			//trying to add a second step here
 			//if the userRole is partner then redirect to agency.ejs then profile.ejs
-			if (newUser.userRole == 'partner') {
+
+			const emailResponse = await sendMail('gnorbsl@gmail.com', email, 'Email verification', 'Please verify your email by clicking on this link: http://localhost:8081/users/verify/' + verificationHash)
```
Comment (Jacobjeevan) [inline routes/users.js:235]: Don't forget to swap your email for user's email :)

Comment (stacysealky) [reply routes/users.js:235]: @gnorbsl @Jacobjeevan ^ what Jacob said

**Comment by gnorbsl** [comment]
Still needs some work, closing for now

File: controllers/email.js [hunk 2]
```diff
@@ -14,33 +14,33 @@
...
 const mailGun = require('nodemailer-mailgun-transport');
+const bcrypt = require('bcrypt');
 
 //cb is callback, cb(err, null) means if err, get err, else null
 //IF WE WANT TO CHANGE THE RECIPIENT ADDRESS LATER, MUST AUTHORIZE IN MAILGUN SYSTEM FIRST
-const sendMail = (email, name, subject, message, cb) => {
-	
+const sendMail = async (from, to, subject,  message) => {
+
 	getTransport()
```
Comment (Tokikko) [inline controllers/email.js:23]: Can we write this as async/await instead of a promise?

Comment (gnorbsl) [reply controllers/email.js:23]: Yes fixed it in the next version

**Pull request closed without merging**